utils.go

```diff
@@ -16,18 +17,28 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	// First, look for a PROPAGATION_FORMAT_GO_HTTP_HEADER injector (our preference).
+	injector := sp.PropagationInjectorForFormat(PROPAGATION_FORMAT_GO_HTTP_HEADER)
```

Comment:
@yurishkuro @tschottdorf I should explain this more... @yurishkuro, I agree that the formats declared within OpenTracing proper are mandatory. This example was meant to illustrate what some off-brand RPC library could do with their *own* format (i.e., use it if available, but fall back to something standard if not). So, imagine for the sake of illustration that the GO_HTTP_HEADER format is non-standard in this example.